tests/rptest/tests/datalake/schema_evolution_test.py

```diff
@@ -20,210 +29,560 @@
-                                   cloud_storage_enable_remote_write=False),
-            extra_rp_conf={
-                "iceberg_enabled": True,
-                "iceberg_catalog_commit_interval_ms": 5000,
+        self._generate_record = generate_record
+        self._spark_table = spark_table
+        self._trino_table = trino_table
+
+    def load(self):
+        return avro.loads(json.dumps(self._rep))
+
+    def table(self, engine: QueryEngineType):
+        return self._spark_table if engine == QueryEngineType.SPARK else self._trino_table
+
+    @property
+    def field_names(self) -> list[str]:
+        return [field['name'] for field in self._rep['fields']]
+
+    @property
+    def fields(self) -> list[dict]:
+        return self._rep['fields']
+
+    def gen(self, x: float) -> dict:
+        return self._generate_record(x)
+
+
+class EvolutionTestCase(NamedTuple):
+    initial_schema: AvroSchema
+    next_schema: AvroSchema
+
+
+LEGAL_TEST_CASES = {
+    "add_column":
+    EvolutionTestCase(
+        initial_schema=AvroSchema(
+            fields=[
+                {
+                    "name": "verifier_string",
+                    "type": "string",
+                },
+            ],
+            generate_record=lambda x: {
+                "verifier_string": f"verify-{x}",
             },
-            schema_registry_config=SchemaRegistryConfig())
-
-    def avro_stream_config(self, topic, subject, mapping: str):
-        return {
-            "input": {
-                "generate": {
-                    "mapping": mapping,
-                    "interval": "",
-                    "count": 1000,
-                    "batch_size": 1
-                }
+            spark_table=[
+                ('verifier_string', 'string'),
+            ],
+            trino_table=[
+                ('verifier_string', 'varchar'),
+            ],
+        ),
+        next_schema=AvroSchema(
+            fields=[
+                {
+                    "name": "verifier_string",
+                    "type": "string",
+                },
+                {
+                    "name": "ordinal",
+                    "type": "int",
+                },
+            ],
+            generate_record=lambda x: {
+                "verifier_string": f"verify-{x}",
+                "ordinal": int(x),
             },
-            "pipeline": {
-                "processors": [{
-                    "schema_registry_encode": {
-                        "url": self.redpanda.schema_reg().split(",")[0],
-                        "subject": subject,
-                        "refresh_period": "10s"
-                    }
-                }]
+            spark_table=[
+                ('verifier_string', 'string'),
+                ('ordinal', 'int'),
+            ],
+            trino_table=[
+                ('verifier_string', 'varchar'),
+                ('ordinal', 'integer'),
+            ],
+        ),
+    ),
+    "drop_column":
```

Comment:
I still think this is too aggressive of behavior, I rarely see people dropping columns and old data in analytical datasets, but that's not really a discussion for this PR.

Reply:
I think #2 and ideally long term we gracefully having the user manually drop the column

Reply:
Interesting, yeah. Variations on the theme of strict vs flexible interpretation of record schema. 

I don't have strong intuition one way or the other. Are we collecting open questions like this somewhere? If not I can start a fresh doc